src/elements/walletConnect/WalletConnectButton.tsx

```diff
@@ -1,26 +1,26 @@
 import { UseWalletConnect } from './useWalletConnect';
-import { shortenString } from 'utils/pureFunctions';
-import { Image } from 'components/image/Image';
+import { shortenString, truncateString } from 'utils/pureFunctions';
 import { ReactComponent as IconWallet } from 'assets/icons/wallet.svg';
+import useENS from 'hooks/useENS';
+import Davatar from '@davatar/react';
 
 export const WalletConnectButton = ({
   handleWalletButtonClick,
   account,
   selectedWallet,
 }: UseWalletConnect) => {
-  const buttonText = account ? shortenString(account) : 'Connect Wallet';
+  const { ensName } = useENS(account);
+  const buttonText = account
+    ? truncateString(ensName) || shortenString(account)
```

Comment:
shouldn't it be `??` instead of `||` to better handle empty string? Unless it's never the case with ensName.